### ISSUE TITLE
Crashing due to using jar launcher instead of native launcher

### ISSUE THREAD

**Issue by Murazriel** [Darkosto/SevTech-Ages#1720]
## Issue / Bug

Crashing due to using jar launcher
## Steps to Reproduce (for bugs)

1. Load into server,
2. Crash

## Context

I can run the mod just fine on me and my friends' server when using the native launcher, however, that makes it so the engineering book crashes me out after a random amount of time of using it. So I switched to jar launching and now I crash out upon loading into the server.
## Logs

* Client/Server Log:
* Crash Log: https://gist.github.com/Murazriel/1e45aa51d1e24bf071dd7a929a7f7aa3

## Client Information

* Modpack Version: 3.0.6
* Java Version:  1.8.0.161 (64 bit) for jar launching, and the forced .25 one for native
* Launcher Used: twitch (jar launcher setting)

* Memory Allocated: 6gigs

* Server/LAN/Single Player: Server using hamachi connection
* Resourcepack Enabled?: no
* Optifine Installed?: no

**Comment by Murazriel** [comment]
I'm not the one hosting the server so I don't have the server logs. Would the client logs you're mentioning be Curse\Minecraft\Instances\SevTech Ages\logs -> debug.txt file? The crash isn't just on the first join so I don't think its what you mentioned.

**Comment by Murazriel** [comment]
I'm being kicked off to main menu and theres no folder named crash-reports inside of \Curse\Minecraft\Instances\SevTech Ages

**Comment by Murazriel** [comment]
Wasn't using the arguments before, put them in and just loading, will be a few minutes

**Comment by artdude543** [comment]
Is anything being displayed on the game when you are kicked out? Like a Timeout error or IO?

**Comment by Murazriel** [comment]
Yeah there is, i forget which one but I'll find out shortly

**Comment by Murazriel** [comment]
Also to note, even just booting up minecraft right now is causing more debug text logs to be created

**Comment by Murazriel** [comment]
using tthe new arguments I  still get Connection Lost - Timed Out. This is what the server says got the host to paste it to me. https://pastebin.com/hKbzUJ0Z

**Comment by artdude543** [comment]
Okay now add this to the args I've already given you `-fml.readTimeout=120` (This will tell the game to wait longer before timing out)

**Comment by artdude543** [comment]
This timeout error is known and we are working on it, but it's one of the rare case issues which we can't reproduce enough to work on trying to fix it. But modders are looking into it but we have no ETA on any kind of fix for it.

**Comment by Murazriel** [comment]
Alright, Rebooting it'll be another 5ish minutes

**Comment by Murazriel** [comment]
Adding that argument to the end of the argument line previously given causes a java virtual machine error.

Java HotSpot(TM) 64-Bit Server VM warning: ignoring option PermSize=256m; support was removed in 8.0
Unrecognized option: -fml.readTimeout=120

**Comment by artdude543** [comment]
Oh wait kill the game @Murazriel I derped with the copy paste >.> Should be `-Dfml.readTimeout=180` (120 is fine if you want to keep that)

**Comment by Murazriel** [comment]
Ah okay, booting up now

**Comment by Murazriel** [comment]
Timeout again, 
https://pastebin.com/yReskP06 server pastebin

https://gist.github.com/Murazriel/491e9d7c4fa6573282f417d0ad0e7b50 debug gist, this is maybe 1/30th of the lines that it spit out

**Comment by artdude543** [comment]
Okay thanks for all that! I'm going to assume this is an issue with the scanable mod which is flooding your client causing the time out.

**Comment by Murazriel** [comment]
So should I just switch back to native launch and just use online resources to explore immersive engineering since the book will crash me using native instead of jar?

**Comment by artdude543** [comment]
Okay this was already fixed in a newer version of the mod https://github.com/MightyPirates/Scannable/issues/38

If can I know this is a long shot if you can update your client tempoary and the sevrer with version 1.6.3 of the mod can you try to get on the sever then?

**Comment by Murazriel** [comment]
I believe I have been using 1.6.3? https://cdn.discordapp.com/attachments/141561674678272000/433599633323720704/unknown.png

**Comment by artdude543** [comment]
Hmmmm okay, Darko never included in the changelog hence I missed it _(would have been nice to know) lol_

Right well, in that case, it's still an issue. I'll post on the issue to get it looked into.

**Comment by Murazriel** [comment]
For a comparison my server admin also switched from native to jar launching to get the book to not crash, maybe there'll be something

My server log crash: https://pastebin.com/6RFVvaFj
His server log crash: https://pastebin.com/aU9gKLnY

**Comment by artdude543** [comment]
I don't think this is a launcher issue tbh; but we found one issue so let's get that one looked into before we pull apart the client to look into other causes 😄

**Comment by Bendder** [comment]
Just an FYI but the -Dfml.readTimeout parameter needs to be set on both the Server and the Client.  I had reported an issue regarding the need for this parameter to be used.  [https://github.com/Darkosto/SevTech-Ages/issues/1315](url)
